Model/Indexer/LuceneSearch.php

```diff
@@ -49,35 +52,51 @@ public function __construct(
      */
     public function prepare(): void
     {
-        $client = $this->connectionManager->getConnection();
         $value = $this->getContext()->getRequestParam('search');
-        $indexPrefix = $this->config->getIndexPrefix();
 
-        try {
-            $rawResponse = $client->query(
-                [
-                    'index' => $indexPrefix . '_async_event_*',
-                    'q' => $value,
-                    // the default page size is 10. The highest limit is 10000. If we want to traverse further, we will
-                    // have to use the search after parameter. There are no plans to implement this right now.
-                    'size' => 100
-                ]
-            );
+        if ($this->asyncEventsConfig->isIndexingEnabled()) {
```

Comment:
In both the `if` and the `else` cases, you're only doing anything if `$value` is not empty. I'd suggest taking that out as an early return condition:
```
if (empty($value)) {
    return;
}

if ($this->asyncEventsConfig->isIndexingEnabled()) {
    // lucene search
} else {
    // apply basic 'like' filter
}
```

Reply:
i've updated as suggested :+1: